### DISCUSSION TITLE
How to add 'Site Template' to template 'From Your Organization' section using PnP.PowerShell

### DISCUSSION THREAD

**Discussion by MAP-Git** [pnp/powershell#1182] [Q&A]
![Site-Template](https://user-images.githubusercontent.com/66069197/135506191-ba5ffb68-9582-4d29-8a38-ff6de6b7b621.png)

How to add 'Site Template' to template 'From Your Organization' section using PnP.PowerShell?
As per the attachment, we have to add Site Template to 'From Your Orgnization' section. 
Please help us with commands and steps.

**Comment by Louicha** [comment]
Hi @MAP-Git, 
I will try to summarize it for you in a few steps.
1. Create the site script in JSON. You can do this manually or by using the following commands  [Get-PnPSiteScriptFromList](https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/get-pnpsitescriptfromlist?view=sharepoint-ps) or [Get-PnPSiteScriptFromWeb](https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/get-pnpsitescriptfromweb?view=sharepoint-ps) 
2. Add the site script [Add-PnPSiteScript](https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/add-pnpsitescript?view=sharepoint-ps) and copy the id of the script
3. Create the site template  [Add-PnPSiteDesign](https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/add-pnpsitedesign?view=sharepoint-ps). You can also add a preview image here.

Done, you can see it now under "From your organization". 🎉
If you are using [Add-PnPSiteDesignFromWeb](https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/add-pnpsitedesignfromweb?view=sharepoint-ps)  you can create a new template from an existing site and you only need one command, because steps 1-3 are combined.
 
 More information: 
 [Get started creating site templates and site scripts - MS Docs](https://docs.microsoft.com/en-us/sharepoint/dev/declarative-customization/get-started-create-site-design) 
 
 Hope I was able to help you.
